packages/murmur2/test/hash.t

```diff
@@ -79,6 +79,34 @@
   $ ./compare.sh "z-index: 10"
   Hashes match: 7au0g0
 
+Special chars
```

Comment:
I asked chatgpt for more examples of tests (I don't think the "long strings" one will compile 😓 but still useful):

Adding more targeted tests can ensure that edge cases and potential boundary issues are well-covered. Here are some specific examples of tests to add:

### 1. **Boundary Tests for UTF-16 Code Units**
   - These tests would ensure the code correctly handles characters around the boundary between single UTF-16 code units and surrogate pairs.

   ```ocaml
   $ ./compare.sh 'content: "\u{FFFF}"' (* Last code point within the BMP *)
   Hashes match: expected_hash

   $ ./compare.sh 'content: "\u{10000}"' (* First code point requiring surrogate pairs *)
   Hashes match: expected_hash
   ```

### 2. **Surrogate Pair Ranges**
   - Include specific tests for both high and low surrogates to verify correct handling of surrogate pairs, particularly ensuring they combine correctly.

   ```ocaml
   $ ./compare.sh 'content: "\u{D800}"' (* High surrogate, standalone (should not occur normally) *)
   $ ./compare.sh 'content: "\u{DC00}"' (* Low surrogate, standalone (should not occur normally) *)
   ```

### 3. **Multi-Character Strings with Mixed BMP and Non-BMP Characters**
   - Testing strings that mix BMP characters (e.g., ASCII, accented letters) and characters that require surrogate pairs helps confirm consistent encoding and hashing across different character ranges.

   ```ocaml
   $ ./compare.sh 'content: "a𐍈z"' (* ASCII, non-BMP character, ASCII *)
   Hashes match: expected_hash

   $ ./compare.sh 'content: "é😀中"' (* Accented character, emoji, CJK character *)
   Hashes match: expected_hash
   ```

### 4. **Edge Cases for Non-Printable ASCII and Control Characters**
   - Testing with control characters (e.g., newline, tab) ensures they’re handled appropriately in UTF-16 encoding.

   ```ocaml
   $ ./compare.sh 'content: "Line1\nLine2"' (* Newline in string *)
   Hashes match: expected_hash

   $ ./compare.sh 'content: "Tab\tSpace"' (* Tab character *)
   Hashes match: expected_hash
   ```

### 5. **Long Strings**
   - Test long strings to validate that your code handles lengthy input without performance issues or memory problems, especially important for a hashing algorithm.

   ```ocaml
   $ ./compare.sh 'content: "a" ^ 1000' (* Repeat a single character 1000 times *)
   Hashes match: expected_hash

   $ ./compare.sh 'content: "😀" ^ 500' (* Repeat a surrogate pair character 500 times *)
   Hashes match: expected_hash
   ```

### 6. **Combination of Diacritical Marks and Composite Characters**
   - Certain characters can be combined with diacritical marks to form composite characters. Testing these combinations ensures accurate encoding and hashing.

   ```ocaml
   $ ./compare.sh 'content: "e\u0301"' (* 'e' + acute accent (é) *)
   Hashes match: expected_hash

   $ ./compare.sh 'content: "n\u0303o\u0301"' (* 'n' + tilde, 'o' + acute accent (ñó) *)
   Hashes match: expected_hash
   ```

### 7. **Emoji Sequences and Skin Tone Modifiers**
   - Emojis with skin tone or gender modifiers form composite characters in UTF-16, which could affect hashing.

   ```ocaml
   $ ./compare.sh 'content: "👍🏽"' (* Thumbs up with medium skin tone *)
   Hashes match: expected_hash

   $ ./compare.sh 'content: "👩‍💻"' (* Woman technologist emoji, sequence with zero-width joiner *)
   Hashes match: expected_hash
   ```

### 8. **CJK and Other Multi-Byte Characters**
   - Characters from CJK languages and other multi-byte character sets are crucial to test as they are common in non-Latin scripts and can have varied byte-length representations in UTF-16.

   ```ocaml
   $ ./compare.sh 'content: "漢字"' (* Chinese characters *)
   Hashes match: expected_hash

   $ ./compare.sh 'content: "हिन्दी"' (* Hindi script *)
   Hashes match: expected_hash
   ```

These examples aim to cover a wide variety of characters, including edge cases for surrogate pairs, multi-byte characters, and sequences. They will help validate that the hashing algorithm performs consistently with JavaScript’s UTF-16 expectations across diverse inputs.